controllers/platform/services/services.go

```diff
@@ -358,14 +366,35 @@ func (j JobServiceHandler) MergeContainerSpec(containerSpec *corev1.Container) (
 	return c, err
 }
 
+// hasPostgreSQLConfigured returns true when either the SonataFlow Platform PostgreSQL CR's structure or the one in the Job service specification is not nil
+func (j JobServiceHandler) hasPostgreSQLConfigured() bool {
+	return j.IsServiceSetInSpec() &&
+		((j.platform.Spec.Services.JobService.Persistence != nil && j.platform.Spec.Services.JobService.Persistence.PostgreSQL != nil) ||
+			(j.platform.Spec.Persistence != nil && j.platform.Spec.Persistence.PostgreSQL != nil))
+}
+
 func (j JobServiceHandler) ConfigurePersistence(containerSpec *corev1.Container) *corev1.Container {
 
-	if j.platform.Spec.Services.JobService.Persistence != nil && j.platform.Spec.Services.JobService.Persistence.PostgreSql != nil {
+	if j.hasPostgreSQLConfigured() {
+		var p *operatorapi.PersistencePostgreSQL
 		c := containerSpec.DeepCopy()
 		c.Image = j.GetServiceImageName(constants.PersistenceTypePostgreSQL)
-		c.Env = append(c.Env, persistence.ConfigurePostgreSqlEnv(j.platform.Spec.Services.JobService.Persistence.PostgreSql, j.GetServiceName(), j.platform.Namespace)...)
+		if j.platform.Spec.Services.JobService.Persistence != nil && j.platform.Spec.Services.JobService.Persistence.PostgreSQL != nil {
+			p = j.platform.Spec.Services.JobService.Persistence.PostgreSQL
+		} else {
+			p = &operatorapi.PersistencePostgreSQL{
+				SecretRef: j.platform.Spec.Persistence.PostgreSQL.SecretRef,
+				JdbcUrl:   j.platform.Spec.Persistence.PostgreSQL.JdbcUrl,
+			}
+			if j.platform.Spec.Persistence.PostgreSQL.ServiceRef != nil {
+				p.ServiceRef = &operatorapi.PostgreSQLServiceOptions{
+					SQLServiceOptions: j.platform.Spec.Persistence.PostgreSQL.ServiceRef,
+					DatabaseSchema:    j.GetServiceName()}
+			}
+		}
```

Comment:
Any reason `persistence.RetrieveConfiguration` is not also being used here?

Reply:
Done.